### DISCUSSION TITLE
How could I change RouterView based on params?

### DISCUSSION THREAD

**Discussion by s3xysteak** [vuejs/router#2339] [Help and Questions] [answered]
```
/my/:typeId?/:id?
+-----------------------------------------------------------+ 
| Selector                                                  |             
| +-------------------------------------------------------+ |  
| | id ? `<CompId />` : typeId ? `<CompTypeId/>` : nothing| |  
| +-------------------------------------------------------+ |     
+-----------------------------------------------------------+
```

**Comment by posva** [comment, marked as answer]
The route `/my/:typeId?/:id?` [is a bit ambiguous](https://paths.esm.dev/?p=AAMsIPQgYAuQngAuwLoFDiQwTIDIDyyBCvAACA..&t=/my#) because of the double optional parameter but it's still realistic in some scenarios. You could avoid the _nothing_ by using a nested route:

```ts
{
  path: '/my'
  children: [
    { path: ':typeId?/:id?' },
  ]
}
```
Displaying a different component based on the available params is totally fine.

**Reply by s3xysteak** [reply]
Thank you! I didn't think I could use param at the beginning of the path.

**Reply by posva** [reply]
Here it's because they are nested paths so it prepends the parent's path
